Regression: ward fails on Mac OS X

Unfortunately PR #40 causes ward to fail on OS X:
```
$ make V=1
CPP="gcc -E" ../../hpcgap/ward/bin/addguards2c -cpp  -I. -I../.. -DHAVE_CONFIG_H   -I/Users/mhorn/Projekte/GAP/gap.github/build/hpcgap/extern/install/gc/include -I/Users/mhorn/Projekte/GAP/gap.github/build/hpcgap/extern/install/libatomic_ops/include -Wall -Wextra -Wshadow -Wno-unused-parameter -Wno-sign-compare -Wno-implicit-fallthrough -Wno-unknown-warning-option -isystem /sw/include -- src/ariths.c > gen/ariths.c
/usr/include/mach/i386/_structs.h:92:18: Syntax error
        __denorm :1,
                 ^
ERROR: Ward failed to parse C source
make: *** [../../Makefile.rules:243: gen/ariths.c] Error 1
make: *** Deleting file 'gen/ariths.c'
```

--- a/ward/cparser.lua
+++ b/ward/cparser.lua
@@ -857,7 +857,7 @@ local grammar = pattern {
   type_declaration_init = rule "type_declaration" * rule "opt_asm_declaration" *
     (sp * "=" * sp * rule "init_expression" + value(nil)),
   type_postfix =
-    sp * pattern ":" * sp * expression * value({})+
+    sp * pattern ":" * sp * rule "cond_expression" * value({})+
     aggregate((sp * rule "type_postfix_item")^0),
   type_postfix_item =
     pattern "[" * sp * pattern "]" * value("*") +

